js/wallet/walletModel.js

```diff
@@ -0,0 +1,45 @@
+import Observable from "../common/observable.js";
+
+class WalletModel extends Observable {
+  constructor() {
+    super();
+    this.init();
+    this.walletData = {
+      walletMoney: {},
+      walletTotal: 0,
+      VMCash: 0,
+      currentIndex: null
+    };
+  }
+  async init() {
+    const response = await fetch("../mockdata/walletMoney.json");
+    const data = await response.json();
+    this.walletData.walletMoney = data;
+    await this.setWalletData(data);
+}
+  pay(cash) {
+    for (let i = 0; i < this.walletData.walletMoney.length; i++) {
+      const moneyData = this.walletData.walletMoney[i];
+      if (moneyData.money == cash && moneyData.money_qty > 0) {
+        this.walletData.currentIndex = i;
+        this.walletData.walletTotal -= cash;
+        this.walletData.VMCash += cash;
+        this.walletData.walletMoney[i].money_qty -= 1;
+      }
+    }
+    this.notify(this.walletData);
+  }
+  
+  setWalletData(initialData) {
+    this.walletData.walletMoney = initialData;
+    const walletTotal = initialData.reduce(
+      (total, data) => (total += data.money * data.money_qty),
```

Comment:
(Array.reduce)[https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce]에서 total이 누적되므로 아래처럼 해야하는 것이 아닌가요?
```suggestion
      (total, data) => (total + data.money * data.money_qty),
```